Editor/Data/Attribute/RestrictedAttributeDrawer.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Zinnia.Data.Attribute
+{
+    using UnityEngine;
+    using UnityEditor;
+
+    [CustomPropertyDrawer(typeof(RestrictedAttribute))]
+    public class RestrictedAttributeDrawer : PropertyDrawer
+    {
+        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        {
+            EditorGUI.BeginProperty(position, label, property);
+            RestrictedAttribute attrib = attribute as RestrictedAttribute;
+
+            bool playingAndActiveAndEnabled = Application.isPlaying && property.serializedObject.targetObject is Behaviour behaviour && behaviour.isActiveAndEnabled;
+            bool makeReadOnly = (attrib.restrictions & RestrictedAttribute.Restrictions.ReadOnlyAlways) != 0 ||
+                ((attrib.restrictions & RestrictedAttribute.Restrictions.ReadOnlyWhenPlayingAndDisabled) != 0 && playingAndActiveAndEnabled);
+
+            bool muteProperty = (attrib.restrictions & RestrictedAttribute.Restrictions.Muted) != 0;
+
+            bool originalGUIEnabledState = GUI.enabled;
+            if (makeReadOnly)
+            {
+                GUI.enabled = false;
+            }
+
+            Color controlColor = Color.grey;
+
+            Color currentGUIColor = GUI.color;
+            FontStyle currentFontStyle = EditorStyles.label.fontStyle;
+            Color currentNormalTextColor = EditorStyles.label.normal.textColor;
+            Color currentFocusedTextColor = EditorStyles.label.focused.textColor;
+
+            if (muteProperty)
+            {
+                GUI.color = controlColor;
+                EditorStyles.label.normal.textColor = controlColor;
+                EditorStyles.label.focused.textColor = controlColor;
+                EditorStyles.label.fontStyle = FontStyle.Italic;
+            }
+
+            EditorGUI.PropertyField(position, property, label, true);
+
+            if (muteProperty)
+            {
+                GUI.color = currentGUIColor;
+                EditorStyles.label.normal.textColor = currentNormalTextColor;
+                EditorStyles.label.focused.textColor = currentFocusedTextColor;
+                EditorStyles.label.fontStyle = currentFontStyle;
+            }
+
+            if (makeReadOnly)
+            {
+                GUI.enabled = originalGUIEnabledState;
+            }
```

Comment:
```suggestion
            }

```
I feel like we always split control statements with the following expressions with some space like this, but your call!